### PR TITLE
Update developer guide for Shortcuts feature

### LINKED ISSUE #100: Duplicate shortcuts bug
Adding duplicate shortcut shows a success message. While functionality is unaffected, it is better if the user is notified that they just tried to add an existing shortcut.

image

### PR DESCRIPTION
**Pull request by alvinlim277**
Added shortcuts feature implementation details to the Developer Guide and made minor code improvements.

(Hopefully) fixes #100

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteShortcutCommand.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public CommandResult execute(Model model, CommandHistory commandHistory) throws
         requireNonNull(model);
         StringBuilder message = new StringBuilder();
         shortcutAliasList.forEach(shortcutAlias -> {
-            String feedback = model.getShortcutSettings().removeShortcut(shortcutAlias);
+            String feedback = model.removeShortcut(shortcutAlias);
```
Comment (vansh284) [inline src/main/java/seedu/address/logic/commands/DeleteShortcutCommand.java:43]: Good catch! Making use of the law of demeter makes the code quality better

File: docs/diagrams/ShortcutSettingsClassDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Class AddressBookParser LOGIC_COLOR
+Class "<<interface>>\nModel" as Model
+Class ModelManager
+Class UserPrefs
+Class ShortcutSettings
+Class GuiSettings
+
+AddressBookParser -[LOGIC_COLOR]right-> "1" ModelManager
+ModelManager .up.|> Model
+ModelManager *-right-> "1" UserPrefs
```
Comment (vansh284) [inline docs/diagrams/ShortcutSettingsClassDiagram.puml:16]: I'm not sure about the use of composition in this case. Also, it seems inconsistent with a class diagram earlier on in the DG where there is no composition between `ModelManager` and `UserPrefs`.  
An argument can certainly be made both ways, but I think it's better to leave out composition in cases where we are not sure.

**Review by vansh284 — COMMENTED** [review]
Overall good update on the DG, and fixing of the Shortcuts bug. Just slightly hesitant on the use of composition in one case.

File: docs/diagrams/ShortcutSettingsClassDiagram.puml (hunk 2 above)
Comment (alvinlim277) [reply docs/diagrams/ShortcutSettingsClassDiagram.puml:16]: Good point, I will remove the composition here then!

**Review by vansh284 — APPROVED** [review]
LGTM

**Pull request merged**